tenacity/__init__.py

```diff
@@ -499,7 +499,7 @@ def retry(func: WrappedFn) -> WrappedFn:
 
 @t.overload
 def retry(
-    sleep: t.Callable[[t.Union[int, float]], None] = sleep,
+    sleep: t.Callable[[t.Union[int, float]], t.Union[t.Awaitable[None], None]] = sleep,
```

Comment:
```suggestion
    sleep: t.Callable[[t.Union[int, float]], t.Optional[t.Awaitable[None]] = sleep,
```

Reply:
oh yeah of course, I've gotten too used to `from __future__ import annotations` and just doing `| None`